### PR TITLE
unix,win: add uv_os_{get,set}priority()

### PR DESCRIPTION
**Pull request by cjihrig**

### CODE REVIEW

File: test/test-process-priority.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+
+TEST_IMPL(process_priority) {
+#if defined(__MVS__)
+  RETURN_SKIP("functionality not supported on zOS");
```
[the remark this answers is not in the document]
Comment (cjihrig) [reply test/test-process-priority.c]: It was returning ENOTSUP.

File: docs/src/misc.rst [hunk 2]
```diff
@@ -517,3 +517,31 @@ API
...
+
+    Retrieves the scheduling priority of the process specified by `pid`. The
+    returned value of `priority` is between -20 (high priority) and 19 (low
+    priority).
+
+    .. note::
+        On Windows, the returned priority will equal one of the `UV_PRIORITY`
+        constants.
+
+    .. versionadded:: REPLACEME
```
Comment (santigimeno) [inline docs/src/misc.rst]: Good idea. Should we add smthg to https://github.com/libuv/libuv-release-tool?

**Comment by richardlau** [comment]
> LGTM. Wasn't there an issue somewhere where this was discussed?

https://github.com/nodejs/node/pull/21675

File: test/test-process-priority.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jBarz) [reply test/test-process-priority.c]: The function is indeed being called on z/OS but returning ENOSYS.

From the documentation:
"If the ENOSYS return code is received, your installation does not support this service. Contact your system administrator if you require activation of this service."

So as @richardlau suggests, try setting priority "0", and if that returns ENOTSUP, then skip test?

**Comment by cjihrig** [comment]
Updated to try to test on zOS.

CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/993/
The only failures on win2012r2-vs2017 and alpine-last-latest-x64 are unrelated.

**Comment by cjihrig** [comment]
Force pushed with lease to replace `REPLACEME` with 1.23.0. New commit is e57e07172eba9df2f8fa0cec051cac0c5cac9124.

**Pull request merged**